integration_tests/src/main/python/date_time_test.py

```diff
@@ -215,6 +215,21 @@ def test_to_unix_timestamp(data_gen, ansi_enabled):
         {'spark.sql.ansi.enabled': ansi_enabled})
 
 
+@pytest.mark.parametrize('time_zone', ["UTC", "UTC+0", "UTC-0", "GMT", "GMT+0", "GMT-0"], ids=idfn)
```

Comment:
Looks like there are more aliases for UTC on https://www.timeanddate.com/time/zones/. Could be interesting to see what happens with 

[EGST](https://www.timeanddate.com/time/zones/egst)
[WET](https://www.timeanddate.com/time/zones/wet)
[WT](https://www.timeanddate.com/time/zones/wt)
[Z](https://www.timeanddate.com/time/zones/z)

Reply:
They are not supported in Java. The supported zone IDs are here: https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#SHORT_IDS

Reply:
Does not seem accurate:
```
scala> Seq("EGST", "WET", "WT", "Z").map(tz => scala.util.Try(java.time.ZoneId.of(tz).normalized())).mkString("\n")
res6: String =
Failure(java.time.zone.ZoneRulesException: Unknown time-zone ID: EGST)
Success(WET)
Failure(java.time.zone.ZoneRulesException: Unknown time-zone ID: WT)
Success(Z)
```

Reply:
Maybe depends on Java version?

Reply:
It's java 8 that we support

Reply:
CI failed on timezone WET, I don't think that timezone is truly equivalent to UTC.  It may be now, but I'm not sure it was in the past.  They're certainly not equivalent according to Java's `ZoneId`, and looking at the tzfiles in /usr/share/zoneinfo, I can see that there appears to be significantly more discontinuities recorded in the WET file than in the Etc/UTC file.

Reply:
`zdump`-ed WET and at the very least it has summer time

Checking symlinks /usr/share/zoneinfo/posix/ should be a good approximation for alias timezones

```
find /usr/share/zoneinfo/posix/Etc -type l -ls | grep -- '-> GMT\|-> UTC'
   370335      0 lrwxrwxrwx   1 root     root            3 Mar 21 13:26 /usr/share/zoneinfo/posix/Etc/GMT-0 -> GMT
   370339      0 lrwxrwxrwx   1 root     root            3 Mar 21 13:26 /usr/share/zoneinfo/posix/Etc/Greenwich -> GMT
   370337      0 lrwxrwxrwx   1 root     root            3 Mar 21 13:26 /usr/share/zoneinfo/posix/Etc/GMT0 -> GMT
   370343      0 lrwxrwxrwx   1 root     root            3 Mar 21 13:26 /usr/share/zoneinfo/posix/Etc/Universal -> UTC
   370345      0 lrwxrwxrwx   1 root     root            3 Mar 21 13:26 /usr/share/zoneinfo/posix/Etc/Zulu -> UTC
   370341      0 lrwxrwxrwx   1 root     root            3 Mar 21 13:26 /usr/share/zoneinfo/posix/Etc/UCT -> UTC
   370333      0 lrwxrwxrwx   1 root     root            3 Mar 21 13:26 /usr/share/zoneinfo/posix/Etc/GMT+0 -> GMT
```

So Zulu is a legitimate alias. As expecte it was an interesting exercise at least for me.

Reply:
"WET" failed on my machine but "Z" passed.

Reply:
Sorry for confusion, my sentence about WET looks incomplete. I was actually confirming that it's not UTC because it deviates from it during summer time
```
diff <(zdump -i /usr/share/zoneinfo/UTC) <(zdump -i /usr/share/zoneinfo/WET) | head
2,3c2,1049
< TZ="/usr/share/zoneinfo/UTC"
< -     -       +00     UTC
---
> TZ="/usr/share/zoneinfo/WET"
> -     -       +00     WET
> 1977-04-03    02      +01     WEST    1
> 1977-09-25    01      +00     WET
> 1978-04-02    02      +01     WEST    1
> 1978-10-01    01      +00     WET
```

Since we checked it manually I don't think including "Z" is required.